worker/uniter/runner/jujuc/relation-ids.go

```diff
@@ -73,7 +74,7 @@ func (c *RelationIdsCommand) Run(ctx *cmd.Context) error {
 	}
 	for _, id := range ids {
 		r, err := c.ctx.Relation(id)
-		if err == nil && r.Name() == c.Name {
+		if err == nil && r.Name() == c.Name && r.Life() == life.Alive {
```

Comment:
I'm not sure this is exactly what we want. As a stop gap we could include the relation is life != dead (alive or dying ok). We need to allow dying relations to be included still.

The root cause issue is that network-get does not exclusively use cached info on the hook context, and it errors if a relation is not found. It does handle dead relations ok.

So the proper fix is to change how network-get works by including all info it needs on the hook context. But there's a lot of server side logic and a large part of the model involved so it's not trivial.

At best this PR will narrow the window where the issue can occur, so it's worth landing from that point of view (so long as the above check is changed to != dead). NB I think juju does not set a relation to dead until all post remove processing is done so ignoring dead relations I think is ok.

Reply:
I updated the check to `!=dead` as suggested and edited the commit msg / PR description to indicate that this is a stop gap fix that only applies to workflows that iterate realtion-ids output to query additional relation information. I will copy this to the bug as well later today.